storage/badger/dkg_keys.go

```diff
@@ -16,7 +18,12 @@ type DKGKeys struct {
 	cache *Cache
 }
 
-func NewDKGKeys(collector module.CacheMetrics, db *badger.DB) *DKGKeys {
+func NewDKGKeys(collector module.CacheMetrics, db *badger.DB) (*DKGKeys, error) {
+
+	err := operation.EnsureSecretDB(db)
```

Comment:
Can't find my comment. Sorry if I asked this question again:

Can we let `EnsureSecretDB` check whether the encryption key is correct? Seems if the encryption key is incorrect, the db can still be initialized, and won't learn it's wrong until there is an query.

Reply:
The database will fail to open if the encryption key isn't correct. Going to add a test case that validates this.

Reply:
Added a test here: https://github.com/onflow/flow-go/pull/1309/commits/aab5d8e29dad7fee323179d22c0ee00205d71b7a